macros.py

```diff
@@ -264,11 +265,11 @@ def SatisfiesALanguageSpecificProperty(self) -> bool:
         return self.MustUseMetaprogrammingToTransform
 
 
-MacroMap = DefaultDict[Macro, Set[Invocation]]
+MacroMap = dict[Macro, Set[Invocation]]
 
 
 @dataclass
 class PreprocessorData:
-    mm: MacroMap = field(default_factory=lambda: defaultdict(set))
+    mm: MacroMap = field(default_factory=dict)
```

Comment:
Why switch to a `dict` here?

Reply:
For our usage (generate_macro_translations), we are just iterating over the keys and values present in the dictionary after MacroMap is constructed. I don't think we do any arbitrary accesses so we shouldn't access a key that doesn't exist.

This also caused a lot of type-checking errors in get_interface_equivalent_preprocessordata as we were not constructing defaultdicts (src_pd, tlna_src_pd, ie_pd)